### PR TITLE
#33- Automate DB creation for dev environment- Aisha Athman Lali

### PR DESCRIPTION
**Pull request by aishaathmanlali**
## Description

This is a PR for automating DB for dev environment. I have included the connection string to a shared DB

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: database/database.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
 	"dev": {
 		"driver": "pg",
-		"user": "postgres",
-		"password": "opensesame",
-		"host": "localhost",
+		"user": "cyf_tms_user",
+		"password": "s5WAaZQcQd4g6ElcUE6B0t4NXOmTNF00",
+		"host": "dpg-cqhc5vdds78s73bd159g-a.virginia-postgres.render.com",
```
Comment (soonick) [inline database/database.json]: Never commit secrets

**Review by soonick — CHANGES_REQUESTED** [review]

**Review by beaumoaj — COMMENTED** [review]
I would expect to see an SQL script here to set up the database containing 
- create table statements
- insert statements
- perhaps with a conditional drop database and create database at the start so anyone can just run it and set up a database.

Once the forms send data into the database then the insert statements are no longer necessary as you can create your own data from the UI but there still needs to be a script with the create table statements

**Comment by soonick** [comment]
@beaumoaj , we have database migrations set up for our database. We just need to run the migrations every time the system is started, and the migrations take care of everything you mentioned.

File: database/migrations/sqls/20240723172844-attachments-table-and-modifications-to-tender-table-up.sql [hunk 2]
```diff
@@ -11,6 +11,9 @@ ADD COLUMN no_of_bids_received INT;
 ALTER TABLE tender
 ALTER COLUMN description TYPE VARCHAR(2000);
 
+ALTER TABLE tender
```
Comment (soonick) [inline database/migrations/sqls/20240723172844-attachments-table-and-modifications-to-tender-table-up.sql]: This doesn't belong in this PR

File: database/database.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-		"user": "postgres",
-		"password": "opensesame",
-		"host": "localhost",
-		"database": "cyf_tms",
-		"port": "5432"
+		"user": "${DB_USER}",
+		"password": "${DB_PASS}",
+		"host": "${DB_HOST}",
+		"database": "${DB_NAME}",
+		"port": "${DB_PORT}"
```
Comment (soonick) [inline database/database.json]: We need to do this for production. For the dev environment, we can leave it the way it was before

File: scripts/create_dev_db.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```
Comment (soonick) [inline scripts/create_dev_db.sh]: We don't need this script. This command already takes care of everything:

```
node ../node_modules/db-migrate/bin/db-migrate up
```

The only thing we need is to make sure this command is automatically called every time we start the dev version of the server, by modifying the correct npm script

**Review by soonick — CHANGES_REQUESTED** [review]

**Review by soonick — APPROVED** [review]

**Pull request merged**